js/ui/search.js

```diff
@@ -333,7 +333,7 @@ const ListSearchResults = new Lang.Class({
     },
 
     _createResultDisplay: function(meta) {
-        return this.parent(meta) || new ListSearchResult(this.provider, meta);
```

Comment:
I think you're right, I'll remove it. Also, I've just noticed a bug in my PR that caused apps launched from the grid not to work because I didn't add the `activateResult()` method in `AppSearchProvider`, which I'm adding now. And while adding that I'll probably can remove the `createResultObject()` method from there as it won't be needed anymore, rendering this change here useless.